public/app/features/dashboard/components/PanelEditor/VisualizationButton.tsx

```diff
@@ -1,45 +1,31 @@
 import React, { FC } from 'react';
 import { css } from '@emotion/css';
-import { GrafanaTheme, PanelPlugin } from '@grafana/data';
+import { GrafanaTheme } from '@grafana/data';
 import { ToolbarButton, ButtonGroup, useStyles } from '@grafana/ui';
 import { StoreState } from 'app/types';
-import { connect, MapStateToProps, MapDispatchToProps } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 import { setPanelEditorUIState, toggleVizPicker } from './state/reducers';
 import { selectors } from '@grafana/e2e-selectors';
 import { PanelModel } from '../../state';
+import { getPanelPluginWithFallback } from '../../state/selectors';
 
-interface OwnProps {
+type Props = {
   panel: PanelModel;
-}
-
-interface ConnectedProps {
-  plugin?: PanelPlugin;
-  isVizPickerOpen: boolean;
-  isPanelOptionsVisible: boolean;
-}
-
-interface DispatchProps {
-  toggleVizPicker: typeof toggleVizPicker;
-  setPanelEditorUIState: typeof setPanelEditorUIState;
-}
-
-type Props = OwnProps & ConnectedProps & DispatchProps;
+};
 
-export const VisualizationButtonUnconnected: FC<Props> = ({
-  plugin,
-  toggleVizPicker,
-  isPanelOptionsVisible,
-  isVizPickerOpen,
-  setPanelEditorUIState,
-}) => {
+export const VisualizationButton: FC<Props> = ({ panel }) => {
   const styles = useStyles(getStyles);
+  const dispatch = useDispatch();
+  const plugin = useSelector(getPanelPluginWithFallback(panel.type));
```

Comment:
nit: I think that React recommends using useDispatch|Selector over connected components. But when I use it I found it a bit more complicated to test those components because I need to either wrap the component in a Provider or mock useDispatch|Selector. What are everyone's thoughts on this, is there another trick for testing these components? Is there a performance penalty using connected components?

Reply:
I think they recommend the hooks mostly for reducing boilerplate, making it easier to follow the flow of data, and easier typing. I was in two minds about it but felt that for the size of this component it felt neater to remove all the boilerplate. For testing I don't think there's a trick you've missed in your approaches I'm afraid. I've never really considered mocking useDispatch/useSelector. I've always gone the wrong of using a wrapper to handle the mocks. Maybe this refactor shouldn't be part of this PR? I can always revert the commit and we can have this discussion with a wider audience. WDYT @hugohaggmark?

Reply:
Yes, I totally agree with you Jack that this should be with a wider audience and not for this PR. I was just being curious if I missed a trick out there. Again great work 🎉

Reply:
@hugohaggmark could this be part of your useDispatch|Selector topic on next FE Weekly?

Reply:
Hmm not really, that session was more about using useReducer and Redux Toolkit